syntax = "proto3";

import "google/protobuf/empty.proto";

package MyPackage;

service Lamp{
    rpc getState(google.protobuf.Empty) returns (LampState) {}
    rpc setState(LampState) returns (google.protobuf.Empty) {}
}

message LampState{
    enum State{
        OFF = 0;
        ON = 1;
    }
    State state = 1;
}

//python3 -m grpc_tools.protoc -I./proto --python_out=. --grpc_python_out=. ./proto/lamp.proto